packages/govern-subgraph/src/GovernQueue.ts

```diff
@@ -11,148 +11,182 @@ import {
 
 export function handleScheduled(event: ScheduledEvent): void {
   const queue = loadOrCreateQueue(event.address)
-
-  const packet = loadOrCreateQueuePacket(event.params.containerHash, event)
-
+  const payload = loadOrCreatePayload(event.params.containerHash)
+  const container = loadOrCreateContainer(event.params.containerHash, event)
+  // Builds each of the actions bundled in the payload,
+  // and saves them to the DB.
   buildActions(event)
-
-  packet.status = SCHEDULED_STATUS
-  packet.nonce = event.params.payload.nonce
-  packet.executionTime = event.params.payload.executionTime
-  packet.submitter = event.params.payload.submitter
-  packet.executor = event.params.payload.executor.toHexString()
-  packet.allowFailuresMap = event.params.payload.allowFailuresMap
-  packet.proof = event.params.payload.proof
-
-  const schedule = new ScheduleEntity(event.params.containerHash.toHexString())
-
-  const scheduleDeposit = loadOrCreateCollateral(event, '1')
-  scheduleDeposit.token = buildERC20(event.params.collateral.token)
-  scheduleDeposit.amount = event.params.collateral.amount
-
-  schedule.packet = packet.id
-  schedule.collateral = scheduleDeposit.id
-  schedule.createdAt = event.block.timestamp
-
-  // add the packet
-  const scheduled = queue.scheduled
-  scheduled.push(schedule.id)
-  queue.scheduled = scheduled
-
-  packet.save()
-  schedule.save()
+  payload.nonce = event.params.payload.nonce
+  payload.executionTime = event.params.payload.executionTime
+  payload.submitter = event.params.payload.submitter
+  payload.executor = event.params.payload.executor.toHexString()
+  payload.allowFailuresMap = event.params.payload.allowFailuresMap
+  payload.proof = event.params.payload.proof
+  container.payload = payload.id
+
+  const amountOfEvents = container.history.length
+  const containerEvent = loadOrCreateContainerEvent(
+    event.params.containerHash,
+    amountOfEvents
+  )
+  const containerEventId = buildActionId(
+    event.params.containerHash,
+    amountOfEvents
+  )
+  containerEvent.createdAt = event.block.timestamp
+  containerEvent.type = SCHEDULE_CONTAINER_EVENT
+  containerEvent.data = []
+  container.history = [containerEventId]
+
+  container.state = SCHEDULED_STATUS
+  // This should always be possible, as a queue without a config
+  // should be impossible to get at this stage
+  container.config = queue.config
+  container.queue = queue.id
+
+  // add the container to the queue
+  const scheduled = queue.queued
+  scheduled.push(container.id)
+  queue.queued = scheduled
+
+  containerEvent.save()
+  payload.save()
+  container.save()
   queue.save()
 }
 
 export function handleExecuted(event: ExecutedEvent): void {
-  const packet = loadOrCreateQueuePacket(event.params.containerHash, event)
-
-  packet.status = EXECUTED_STATUS
+  const container = loadOrCreateContainer(event.params.containerHash, event)
+  container.state = EXECUTED_STATUS
 
-  packet.save()
+  container.save()
```

Comment:
Should we create a `containerEvent.type = EXECUTE_CONTAINER_EVENT` here?

Also should we have container events for:

- `handleResolved`
- `handleEvidenceSubmitted`
- `handleRuled`

And remove the ones we are not using from the list?